### PR TITLE
Allow the AI to turn on groundside generators like shipside ones, provided that they are intact.

### PR DESCRIPTION
**Pull request by lbnesquik**
## About ThePull Request

This also adds an intact generator. You might want for a map or saving time on testing sometimes.

## Why It's Good For The Game
More consistent :)

## Changelog
:cl:
qol:  the AI can interact with colony generators :)
/:cl:

### CODE REVIEW

File: code/modules/power/groundmap_geothermal.dm [hunk 1]
```diff
@@ -309,6 +312,11 @@
 	name = "\improper Reinforced Reactor Turbine"
 	is_corruptible = FALSE
 
+///This one spawns intact
+/obj/machinery/power/geothermal/intact
+	corrupted = 0
+	buildstate = GEOTHERMAL_NO_DAMAGE
```
Comment (TiviPlus) [inline code/modules/power/groundmap_geothermal.dm:318]: ???

File: code/modules/power/groundmap_geothermal.dm [hunk 2]
```diff
@@ -192,6 +192,9 @@
 	start_processing()
 	return TRUE
 
+/obj/machinery/power/geothermal/attack_ai(mob/living/silicon/ai/user)
+	attack_hand(user)
```
Comment (TiviPlus) [inline code/modules/power/groundmap_geothermal.dm:196]: Do not do this, make them use a shared proc or such instead

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/power/groundmap_geothermal.dm (hunk 1 above)
Comment (lbnesquik) [reply code/modules/power/groundmap_geothermal.dm:322]: Added for the sake of more tools in our toolbox.

File: code/modules/power/groundmap_geothermal.dm (hunk 2 above)
Comment (lbnesquik) [reply code/modules/power/groundmap_geothermal.dm]: Done

File: code/modules/power/groundmap_geothermal.dm [hunk 3]
```diff
@@ -309,6 +316,11 @@
 	name = "\improper Reinforced Reactor Turbine"
 	is_corruptible = FALSE
 
+///This one spawns intact
+/obj/machinery/power/geothermal/intact
+	corrupted = 0
```
[the remark this answers is not in the document]
Comment (lbnesquik) [reply code/modules/power/groundmap_geothermal.dm:321]: Fixed

File: code/modules/power/groundmap_geothermal.dm [hunk 4]
```diff
@@ -309,6 +316,11 @@
 	name = "\improper Reinforced Reactor Turbine"
 	is_corruptible = FALSE
 
+///This one spawns intact
+/obj/machinery/power/geothermal/intact
+	corrupted = ""
+	buildstate = GEOTHERMAL_NO_DAMAGE
```
Comment (TiviPlus) [inline code/modules/power/groundmap_geothermal.dm:322]: delete it if it isnt used

File: code/modules/power/groundmap_geothermal.dm [hunk 5]
```diff
@@ -153,15 +153,22 @@
...
-/obj/machinery/power/geothermal/attack_hand(mob/living/user)
+/obj/machinery/power/geothermal/attack_hand(mob/living/carbon/user)
+	interact_hand(user)
+
+/obj/machinery/power/geothermal/attack_ai(mob/living/silicon/ai/user)
+	interact_hand(user)
+
+
+/obj/machinery/power/geothermal/proc/interact_hand(mob/living/user)
 	. = ..()
```
Comment (TiviPlus) [inline code/modules/power/groundmap_geothermal.dm:164]:
```suggestion
/obj/machinery/power/geothermal/proc/interact_hand(mob/living/user)
```
will fix linter

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/power/groundmap_geothermal.dm (hunk 4 above)
Comment (lbnesquik) [reply code/modules/power/groundmap_geothermal.dm:322]: I think it was useful, but done.

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**